core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/workflow/WorkflowResource.scala

```diff
@@ -414,22 +428,68 @@ class WorkflowResource extends LazyLogging {
     resultWorkflows.toList
   }
 
+  @GET
+  @Path("/content/{wid}")
+  def getWorkflowContentByWid(@PathParam("wid") wid: UInteger): String = {
+    val workflow = context
+      .selectFrom(WORKFLOW)
+      .where(WORKFLOW.WID.eq(wid))
+      .fetchOneInto(classOf[Workflow])
+
+    if (workflow == null) {
+      throw new WebApplicationException(s"Workflow with wid $wid not found.", 404)
+    }
+
+    val objectMapper = new ObjectMapper()
+    val contentMap =
+      objectMapper.readValue(workflow.getContent, classOf[java.util.Map[String, Any]])
+
+    val operators =
+      contentMap.get("operators").asInstanceOf[java.util.List[java.util.Map[String, Any]]]
+    val operatorInfoList = operators.asScala.map { operator =>
+      Map(
+        "operatorID" -> operator.get("operatorID"),
+        "operatorType" -> operator.get("operatorType")
+      ).asJava
+    }.asJava
+
+    objectMapper.writeValueAsString(operatorInfoList)
+  }
+
+//  @GET
+//  @Path("/content/{wid}")
+//  def getWorkflowContentByWid(@PathParam("wid") wid: UInteger): String = {
+//    val workflow = context
+//      .selectFrom(WORKFLOW)
+//      .where(WORKFLOW.WID.eq(wid))
+//      .fetchOneInto(classOf[Workflow])
+//
+//    if (workflow == null) {
+//      throw new WebApplicationException(s"Workflow with wid $wid not found.", 404)
+//    }
+//
+//    workflow.getContent
+//  }
```

Comment:
two questions:
1. why comment it out instead of removing it?
2. why this fix needs to rewrite "getContentByWid"?

Reply:
Since we need the new operator ID passed from the frontend, only the frontend can generate the operator ID now. Previously, cloning didn’t require this.